Flatten project structure

jrayback opened the pull request:
- Flatten project structure by deleting extraneous `vlei-verifier-workflows` folder and moving its contents up one level
- Temporarily replace the broken `run-workflow-test` with a dummy test that always passes
- Set up `eslint` and `prettier`
- Update `.gitignore` to ignore VS Code project files
- Delete`.prettierignore` - it wasn't doing anything since everything is covered in `.gitignore`
- Delete `.npmignore` since `"files": [ "/dist" ]` is set in `package.json`

Fixes #19 #20 #21

jrayback commented:
@2byrds , let me know how you want to handle merging this branch (timing, etc.)...

jrayback commented:
@2byrds , yes. That's what I was thinking too. It's not necessarily a hard merge, but it's a disruptive one for sure. I'll keep my eye out for the green light.

jrayback commented:
@2byrds , do you still want me to hold off on this merge? I see another PR was opened.

2byrds commented:
@jrayback it looks like this PR still has conflicts so I will merge @aydarng smaller PR and then lets review this when it no longer has conflicts

jrayback commented:
@2byrds , ok. But I think #22 will always have conflicts since it is a project restructuring and moves lots of files around. It will probably become harder to merge the longer we wait. Let me know when you're ready!

2byrds commented:
> @2byrds , ok. But I think #22 will always have conflicts since it is a project restructuring and moves lots of files around. It will probably become harder to merge the longer we wait. Let me know when you're ready!

@jrayback once you have fully merged the changes from main (after #11 is merged) then github will no longer list conflicts. You resolve those conflicts during your merge of main into your branch and then github will recognize your the conflicts have been resolved and that your branch is ready to merge as it will include all of those PR changes. Granted your PR will cause a seismic shift by changing the folder structure. But github should not consider that a conflict if you have merged fully.
To summarize, your PR will have all the functionality of main (including #11 changes) but with a new folder structure, some linting, etc.

jrayback commented:
@2byrds , this is ready to go. The project structure is reformatted. I've also fixed linting and Prettier. I prettified all the existing files and added a check in the CI/CD actions to make sure future checkins comply. Let me know if you have any questions.

2byrds commented on test/run-workflow.test.ts:
```diff
@@ -0,0 +1,44 @@
+// import { VleiIssuance } from "../src/vlei-issuance";
```
can this be uncommented now?

jrayback replied:
I'm getting this error: `test/run-workflow.test.ts:38:14 - error TS2339: Property 'prepareClients' does not exist on type 'WorkflowRunner'.` In the definition of WorkflowRunner in `.../utils/run-workflow.ts` I can confirm that that method does not exist on the class. Has someone else fixed this somewhere else?

aydarng replied:
> I'm getting this error: `test/run-workflow.test.ts:38:14 - error TS2339: Property 'prepareClients' does not exist on type 'WorkflowRunner'.` In the definition of WorkflowRunner in `.../utils/run-workflow.ts` I can confirm that that method does not exist on the class. Has someone else fixed this somewhere else?

Hi @jrayback . I removed prepareClients in this [PR](https://github.com/GLEIF-IT/vlei-verifier-workflows/pull/24) and there is no need to call WorkflowRunner.prepareClients() method any more in run-workflow.test.ts.

2byrds reviewed (commented):
Looks great @jrayback. I had one comment for run-workflow.test.ts

jrayback commented:
@2byrds @aydarng , I've taken your feedback into account. The code in `run-workflow.ts`  has been uncommented and matches `main`. Tests are all passing (I guess `docker compose up` needs to run before local tests will pass - I discovered that!). Let me know if there are any other issues with this PR.

aydarng reviewed (approved):
LGTM

2byrds reviewed (approved):
LGTM
